### PR TITLE
Billing: Remove unused code not behind new payment methods flag

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* Once we're confident we won't need to turn off the `purchases/new-payment-methods` flag, we can clean up this code that's only active when the flag is not set.

#### Testing instructions

This PR should only be removing code that is inactive due to the `purchases/new-payment-methods` flag being permanently set in all deployment environments.

We should also make sure the old urls redirect correctly.

- `/me/purchases/:site/:purchaseId/payment/add` to `/me/purchases/:site/:purchaseId/payment-method/add`
- `/me/purchases/add-credit-card` to `/me/purchases/add-payment-method`
- `/me/purchases/:site/:purchaseId/payment/change/:cardId` to `/me/purchases/:site/:purchaseId/payment-method/change/:cardId`

Legacy URLs (not sure if these are used anywhere; it might be worth figuring that out so these can be removed):

- `/purchases/:siteName/:purchaseId/payment/add` to `/purchases/subscriptions/:siteName/:purchaseId/payment-method/add`
- `/purchases/:siteName/:purchaseId/payment/edit/:cardId` to `/purchases/subscriptions/:siteName/:purchaseId/payment-method/change/:cardId`
- `/purchases/subscriptions/:site/:purchaseId/payment/add` to `/purchases/subscriptions/:site/:purchaseId/payment-method/add`
- `/purchases/subscriptions/:site/:purchaseId/payment/edit/:cardId` to `/purchases/subscriptions/:site/:purchaseId/payment-method/change/:cardId`
- `/purchases/add-credit-card/:site` to `/purchases/add-payment-method/:site`

### CODE REVIEW

File: client/me/purchases/index.js [hunk 1]
```diff
@@ -138,47 +132,35 @@ export default ( router ) => {
...
+		makeLayout,
+		clientRender
+	);
 
 	// redirect legacy urls
 	router( '/me/billing', () => page.redirect( paths.billingHistory ) );
 	router( '/me/billing/:receiptId', ( { params: { receiptId } } ) =>
 		page.redirect( paths.billingHistoryReceipt( receiptId ) )
 	);
+	router( paths.addCardDetails( ':site', ':purchaseId' ), ( { params: { site, purchaseId } } ) =>
```
Comment (sirbrillig) [inline client/me/purchases/index.js:158]: 🤔  Since it appears these routes were effectively removed by the flag until now, maybe they're not necessary and/or are covered by something else? Let's check.

File: client/me/purchases/manage-purchase/change-payment-method/index.jsx [hunk 2]
```diff
@@ -93,11 +91,6 @@ function ChangePaymentMethod( props ) {
 		);
 	}
 
-	const recordFormSubmitEvent = () =>
-		void props.recordTracksEvent( 'calypso_purchases_credit_card_form_submit', {
```
Comment (sirbrillig) [inline client/me/purchases/manage-purchase/change-payment-method/index.jsx]: I don't see this event anywhere in the new setup... maybe we should migrate it (this could be a separate PR).

File: client/me/purchases/manage-purchase/change-payment-method/index.jsx [hunk 3]
```diff
@@ -186,13 +164,10 @@ function getCurrentPaymentMethodId( payment ) {
...
 function getChangePaymentMethodTitleCopy( currentPaymentMethodId ) {
-	if ( isEnabled( 'purchases/new-payment-methods' ) ) {
-		if ( [ 'credits', 'none' ].includes( currentPaymentMethodId ) ) {
-			return titles.addPaymentMethod;
-		}
-		return titles.changePaymentMethod;
+	if ( [ 'credits', 'none' ].includes( currentPaymentMethodId ) ) {
+		return titles.addPaymentMethod;
 	}
-	return titles.editCardDetails;
```
Comment (sirbrillig) [inline client/me/purchases/manage-purchase/change-payment-method/index.jsx:142]: I think `titles.editCardDetails` is only used in one other place. Maybe we can remove it entirely? (There may be other titles we can remove with this PR too.)

File: client/me/purchases/paths.js [hunk 4]
```diff
@@ -46,6 +46,7 @@ export function confirmCancelDomain( siteName, purchaseId ) {
 	return managePurchase( siteName, purchaseId ) + '/confirm-cancel-domain';
 }
 
+// legacy path
```
Comment (sirbrillig) [inline client/me/purchases/paths.js:49]: Are these still necessary? Can we replace their remaining uses with the new paths?

File: client/my-sites/purchases/index.js [hunk 5]
```diff
@@ -172,21 +142,27 @@ export default ( router ) => {
...
-						`/purchases/subscriptions/${ siteName }/${ purchaseId }/payment-method/change/${ cardId }`
-				  )
-				: page.redirect(
-						`/purchases/subscriptions/${ siteName }/${ purchaseId }/payment/edit/${ cardId }`
-				  )
+			page.redirect(
+				`/purchases/subscriptions/${ siteName }/${ purchaseId }/payment-method/change/${ cardId }`
+			)
+	);
+	router(
```
Comment (sirbrillig) [inline client/my-sites/purchases/index.js:154]: Similar to my above comment: I'm surprised that we had legacy routes here that have been disabled since the flag was introduced... are we sure they aren't covered elsewhere?

**Review by sirbrillig — COMMENTED** [review]
Everything appears to work correctly. I have some questions about the code but overall I think this is great!

File: client/me/purchases/index.js (hunk 1 above)
Comment (sirbrillig) [reply client/me/purchases/index.js:158]: It's also possible that they are necessary, but were disabled accidentally as a result of our flag and we need to make sure that the legacy routes remain as you've done here; I just wanted to be sure that was the case.

File: client/me/purchases/manage-purchase/change-payment-method/index.jsx (hunk 2 above)
Comment (sirbrillig) [reply client/me/purchases/manage-purchase/change-payment-method/index.jsx]: I think I've actually taken care of this here: https://github.com/Automattic/wp-calypso/pull/48968/commits/690f43c8ba932f9ece1d34c7794cab1b786c226a so you can probably ignore this comment.

**Comment by sirbrillig** [comment]
I removed some of these in https://github.com/Automattic/wp-calypso/pull/49897 as a side-effect of the clean-up there.

**Comment by nbloomf** [comment]
Rebased again

File: client/me/purchases/paths.js (hunk 4 above)
Comment (nbloomf) [reply client/me/purchases/paths.js:49]: I need to make sure I understand the suggestion. If we remove these completely, any old links to (for instance) `/payment/add` will break, right?

**Review by sirbrillig — APPROVED** [review]
This looks fine to me. If there's any more clean-up or further discussion to be had, we can do it in another PR.
